src/HearThis/UI/RecordingToolControl.cs

```diff
@@ -257,52 +272,26 @@ private void HandleScriptBlockRecordingRestored(Project sender, int bookNumber,
 		private Brush[] GetSegmentBrushes()
 		{
 			Guard.AgainstNull(_project, "project");
-			var lineCountForChapter = 0;
-			var expectedCountOfNeededBrushes = 0;
-
-			// We have a so-far unreproducible bug, reported in HT-9,10,35,76,161,161,162, etc.
-			// So we have added a couple of things here until we find the bug:
-			// 1) we try 4 times to get the set of brushes. This increases the chance that the user
-			// can keep working, and, if retries succeed, tells us something about the error.
-			// 2) We passively invite the user to report the problem, using a toast
-			// 3) if retries don't work, we return null and the caller can come up with some fallback approach (e.g. all green)
-			for (int retries = 4; retries > 0; retries--)
-			{
-				try
+			var lineCountForChapter = _project.GetLineCountForChapter(true);
+			var expectedCountOfNeededBrushes = _project.GetLineCountForChapter(!HidingSkippedBlocks);
+			var brushes = new Brush[expectedCountOfNeededBrushes];
+			int iBrush = 0;
+			for (var i = 0; i < lineCountForChapter; i++)
+			{
+				if (GetScriptBlock(i).Skipped)
```

Comment:
Complicated...possibly by all the deleted code. I read:
- if block i is skipped and not hidden, add SkippedSegmentBrush to brushes
- otherwise if we have a clip use blue
- otherwise transparent
So, transparent applies to skipped recordings if we are hiding them, but also to unskipped ones we don't have a clip for. I don't know what these are used for, but that last surprises me...wouldn't we be painting unrecorded blocks we want to record somehow?